ui/src/components/WMWorkerPoolsTable/index.jsx

```diff
@@ -66,23 +91,70 @@ export default class WorkerManagerWorkerPoolsTable extends Component {
     this.setState({ sortBy, sortDirection });
   };
 
+  handleEditClick = ({ currentTarget: { name } }) => {
+    this.props.history.push({
+      pathname: `${this.props.match.path}/${encodeURIComponent(name)}/workers`,
+    });
+  };
+
+  handleDeleteClick = async ({ currentTarget: { name } }) => {
+    const workerPool = this.props.workerPools.find(
+      wp => wp.workerPoolId === name
+    );
+    const payload = {
+      providerId: workerPool.providerId,
+      description: workerPool.description,
+      config: workerPool.config,
+      owner: workerPool.owner,
+      emailOnError: workerPool.emailOnError,
+    };
+
+    this.props.history.replace('/worker-manager');
+
+    try {
+      await this.props.deleteRequest({
+        workerPoolId: workerPool.workerPoolId,
+        payload,
+      });
+    } catch (error) {
+      this.setState({ error: formatError(error), actionLoading: false });
+    }
+  };
+
   renderRow = workerPool => {
     const {
       match: { path },
+      classes,
     } = this.props;
+    const { actionLoading } = this.state;
     const iconSize = 16;
 
     return (
-      <TableRow key={workerPool.workerPool}>
+      <TableRow key={workerPool.workerPoolId}>
+        <TableCell>
+          <IconButton
+            className={classNames(classes.button, classes.editButton)}
+            name={`${workerPool.workerPoolId}`}
+            onClick={this.handleEditClick}
+            disabled={actionLoading || workerPool.providerId === NULL_PROVIDER}>
+            <WorkerIcon size={iconSize} />
+          </IconButton>
+        </TableCell>
+
         <TableCell>
           <TableCellListItem
             button
             component={Link}
-            to={`${path}/worker-pools/${workerPool.workerPoolId}`}>
+            to={`${path}/${encodeURIComponent(workerPool.workerPoolId)}/edit`}>
             <ListItemText
               disableTypography
               primary={<Typography>{workerPool.workerPoolId}</Typography>}
             />
+            {workerPool.providerId === NULL_PROVIDER && (
+              <Label mini status="warning" className={classes.button}>
+                {'To be deleted'}
```

Comment:
I used "To be deleted" because it's shorter and makes for a smaller label. Why is it not clear from the label that the workerpool will be deleted?

Reply:
"Scheduled for Deletion" mentions that there is a plan set-up to delete the resource at a specified later time or when resources become available (e.g, action is queued). "To be deleted" doesn't necessarily means there is a plan. Thoughts?

Reply:
> "To be deleted" doesn't necessarily means there is a plan.

What else can it possibly mean tho? 🤔

Reply:
It can mean it was deleted in the back-end but needs some time for the change to reflect in the UI because of  something like caching. I think adding the word "Scheduled" for example will remove that confusion and better inform the user of what is happening.

Reply:
If it says "_to be_ deleted", it cannot mean it "_was_ deleted", in my opinion